index.js

```diff
@@ -0,0 +1,649 @@
+var bitcoinjs = require('bitcoinjs-lib')
+var BigNumber = require('bignumber.js')
+var _ = require('lodash')
+var encodeAssetId = require('cc-assetid-encoder')
+var cc = require('cc-transaction')
+var findBestMatchByNeededAssets = require('./modules/findBestMatchByNeededAssets')
+
+var CC_TX_VERSION = 0x02
+
+var ColoredCoinsBuilder = function (properties) {
+  properties = properties || {}
+
+  if (typeof properties.network !== 'undefined' && properties.network !== 'testnet' && properties.network !== 'mainnet') {
+    throw new Error('"network" must be either "testnet" or "mainnet"')
+  }
+  this.network = properties.network || 'mainnet' // 'testnet' or 'mainnet'
+
+  if (properties.defaultFee && properties.defaultFeePerKb) {
+    throw new Error('Can have at most one of "defaultFee" and "defaultFeePerKb"')
+  }
+  this.defaultFee = properties.defaultFee
+  this.defaultFeePerKb = properties.defaultFeePerKb
+
+  this.mindustvalue = properties.mindustvalue || 600
+  this.mindustvaluemultisig = properties.mindustvaluemultisig || 700
+  this.writemultisig = properties.writemultisig || true
+}
+
+ColoredCoinsBuilder.prototype.buildIssueTransaction = function (args, callback) {
```

Comment:
`callback` should be removed - all endpoints are synchronous.
(should be done for `buildSendTransaction()` & `buildBurnTransaction()` as well).